### ISSUE TITLE
Set `needs_human_review` on relevant version for some developer actions

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#20508]
https://mozilla-hub.atlassian.net/browse/AMO-47

- When a developer leaves a developer reply, the version the reply was for should receive the `needs_human_review` flag and a due date of `<reply-time>` if that version does not have a due date already.

- When a developer submits source code for a version pending rejection, that version should receive the `needs_human_review` flag and a due date of `<review-time>` if it does not have one already.

**Comment by ioanarusiczki** [comment]
@diox 

- the first scenario worked for me , every time a developer replies the addon will be present in the Manual Review Queue with a due date of 1 day 23 hours (assume it's expected)
- all my attempts for the source code scenario ended in the [pending rejection](https://reviewers.addons-dev.allizom.org/en-US/reviewers/queue/pending_rejection) queue. I did not see the addons present in the Manual Review queue

**Comment by ioanarusiczki** [comment]
@diox 

- I tried with versions pending rejections (signed/unsigned), when source code is uploaded -> they get a human review flag + a due date of <review-time> 

- Same results when I submitted source-code using the endpoint. 

- I did some exploratory so I submitted new versions to see what happens with them. I've noticed that new submitted versions will get a due date when the previous version was rejected. 
A new listed version: https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/625140
A new unlisted verson: https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/625146

They're in the Manual Queue with disabled auto-approval but the queue won't show any icon. 

![backinqueue](https://user-images.githubusercontent.com/33448286/229113604-a664a743-d964-462a-bb52-e439e73dc24b.png)

Let me know if I should file an issue.
